### ISSUE TITLE
Computing loss changes behavior

### ISSUE THREAD

**Issue by mahdiabdollahpour** [Megvii-BaseDetection/YOLOX#1058]
I have a code like below and computing the loss (without backward or step) lowers the accuracy score compared to commenting that whole `once in every N batch` if. the predictions for evaluation are always used from `outputs = model(imgs)` so how can only computing loss without step (so wights no changing) lower the scores?

```
looping over data:
  once in every N batch:
     with torch.cuda.amp.autocast(enabled=self.amp_training):
         model.train()
         loss = model(imgs,targets) # just computing the loss, no backward or step
         model.eval()
  outputs = model(imgs)
  #evaluating
```
